http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcSecurityRealm.java

```diff
@@ -118,7 +118,11 @@ public SupportLevel getEvidenceVerifySupport(Class<? extends Evidence> evidenceT
             }
             RealmAccessClaim realmAccessClaim = accessToken.getRealmAccessClaim();
             if (realmAccessClaim != null) {
-                roles = realmAccessClaim.getRoles().stream().collect(Collectors.toSet());
+                if (roles == null) {
+                    roles = realmAccessClaim.getRoles().stream().collect(Collectors.toSet());
+                } else {
+                    realmAccessClaim.getRoles().stream().forEach(roles::add);
```

Comment:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/Collectors.html#toSet()
```
There are no guarantees on the type, mutability, serializability, or thread-safety of the Set returned
```

Perhaps a good option would be to use:
```java
Set<String> roles = new HashSet<>();

…
roles.addAll(resourceAccessClaim.getRoles());

…
roles.addAll(realmAccessClaim.getRoles());
```

Reply:
I wanted only to create a new `HashSet` if there is actually exist one or both claims, in all other cases a empty set is returned. I changed the implementation to partly go with some of you're suggestions.